app/controllers/api/v1/PostsController.rb

```diff
@@ -0,0 +1,95 @@
+class Api::V1::PostsController < ApplicationController
+  include Authentication
+
+  def index
+    posts = Post.all
+
+    if posts
+      render json: {data: posts}, status: :ok
+    else
+      render json: posts.errors, status: :bad_request
+    end
+  end
+
+  def create
+    if authenticate
+      user = User.find_by(username: request.headers[:username])
+      post = Post.new(title: post_params[:title], description: post_params[:description], slug: "", comments_count: 0, likes_count: 0, user: user)
+
+      if post.save
+        post.update(slug: "api/v1/posts/" << String(post.id))
+        render json: {data: post}, status: 201
+      else
+        render json: post.errors, status: :unprocessale_entity
+      end
+    else
+      render status: 401
+    end
+  end
+
+  def show
+    begin
+    post = Post.find(params[:id])
+      render json: { data: post }, status: :ok
+    rescue ActiveRecord::RecordNotFound
+      render status: :not_found
+    end
+  end
+
+  def destroy
+    if authenticate
```

Comment:
authenticate should be called before create, show, update and destroy. 
So better call it in before action for all these methods.

Reply:
Fixed it

Reply:
Currently I am doing it on every controller individually. Will move it to application_controller